nlopt build problem - CMakeLists.txt missing

This is a new package to my spack build.  I have not added any variant args just ```spack install nlopt```

### Expected Result

clean build

### Actual Result

==> Installing nlopt
curl: (37) Couldn't open file MIRROR_DIR/mirror/nlopt/nlopt-2.4.2.tar.gz
==> Fetching file://MIRROR_DIR/mirror/nlopt/nlopt-2.4.2.tar.gz
==> Fetching from file://MIRROR_DIR/mirror/nlopt/nlopt-2.4.2.tar.gz failed.
==> Fetching https://github.com/stevengj/nlopt/releases/download/nlopt-2.4.2/nlopt-2.4.2.tar.gz
==> Staging archive: WORKING_DIR/var/spack/stage/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/nlopt-2.4.2.tar.gz
==> Created stage in WORKING_DIR/var/spack/stage/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb
==> No patches needed for nlopt
==> Building nlopt [CMakePackage]
==> Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' 'WORKING_DIR/var/spack/stage/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/nlopt-2.4.2' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:PATH=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH:STRING=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/lib64;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/python-2.7.13-52dgd34kwi53hcwhmkggokoplqxzzmfp/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/bzip2-1.0.6-egkg2fbrcljkai2d23pgrq7wzeg2k5p5/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/gdbm-1.14.1-nu73hb3bwwumnr5pe7n2ovwpvn3lj32c/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/readline-7.0-6a33hh42gihqevhhwm7ddsklhu7xwkgk/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/ncurses-6.0-tymhqxlvcy5eduud5ijnohpkuinoi7n4/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/openssl-1.0.2n-b2pwlezjpb63d4lvpnb6hppps7qw4cap/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/zlib-1.2.11-euf4keckd7zopxedqa5pp37smh4v2aa4/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/sqlite-3.18.0-ss65uo3pidaok5xk6ggbrastms2ludx4/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/pcre-8.41-tnraxjn7calk2xer5ructusju54toa5j/lib' '-DCMAKE_PREFIX_PATH:STRING=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/python-2.7.13-52dgd34kwi53hcwhmkggokoplqxzzmfp;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/swig-3.0.12-eww74pu5pubgf2fhsu24t7m227qooxlx;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/cmake-3.7.2-2emsjuf7ycy6ai6l7ln4zkwgybanaug3;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/py-numpy-1.14.2-foa5vlkdxayrm4cv3wrusczloinaxkls' '-DPYTHON_EXECUTABLE=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/python-2.7.13-52dgd34kwi53hcwhmkggokoplqxzzmfp/bin/python2.7'

1 error found in build log:
     1    ==> Executing phase: 'cmake'
     2    ==> 'cmake' 'WORKING_DIR/var/spack/stage/nlopt-2.4.2-wei4zixp
          d7mb2b7zn2uniak5svelp6wb/nlopt-2.4.2' '-G' 'Unix Makefiles' '-DCMAKE_
          INSTALL_PREFIX:PATH=WORKING_DIR/opt/spack/linux-centos7-x86_6
          4/gcc-5.4.0/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb' '-DCMAKE_BU
          ILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-
          DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH
          :STRING=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/
          nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/lib;WORKING_DIR/
          opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-wei4zixpd7mb2b7z
          n2uniak5svelp6wb/lib64;WORKING_DIR/opt/spack/linux-centos7-x8
          6_64/gcc-5.4.0/python-2.7.13-52dgd34kwi53hcwhmkggokoplqxzzmfp/lib;/sc
          1/apps/spack/wip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/bzip2-1.0.6
          -egkg2fbrcljkai2d23pgrq7wzeg2k5p5/lib;WORKING_DIR/opt/spack/l
          inux-centos7-x86_64/gcc-5.4.0/gdbm-1.14.1-nu73hb3bwwumnr5pe7n2ovwpvn3
          lj32c/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.
          0/readline-7.0-6a33hh42gihqevhhwm7ddsklhu7xwkgk/lib;WORKING_DIR
          ip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/ncurses-6.0-tymhqxlvcy5ed
          uud5ijnohpkuinoi7n4/lib;WORKING_DIR/opt/spack/linux-centos7-x
          86_64/gcc-5.4.0/openssl-1.0.2n-b2pwlezjpb63d4lvpnb6hppps7qw4cap/lib;/
          sc1/apps/spack/wip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/zlib-1.2.
          11-euf4keckd7zopxedqa5pp37smh4v2aa4/lib;WORKING_DIR/opt/spack
          /linux-centos7-x86_64/gcc-5.4.0/sqlite-3.18.0-ss65uo3pidaok5xk6ggbras
          tms2ludx4/lib;WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-
          5.4.0/pcre-8.41-tnraxjn7calk2xer5ructusju54toa5j/lib' '-DCMAKE_PREFIX
          _PATH:STRING=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5
          .4.0/python-2.7.13-52dgd34kwi53hcwhmkggokoplqxzzmfp;WORKING_DIR
          ip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/swig-3.0.12-eww74pu5pubgf
          2fhsu24t7m227qooxlx;WORKING_DIR/opt/spack/linux-centos7-x86_6
          4/gcc-5.4.0/cmake-3.7.2-2emsjuf7ycy6ai6l7ln4zkwgybanaug3;WORKING_DIR
          ack/wip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/py-numpy-1.14.2-foa5
          vlkdxayrm4cv3wrusczloinaxkls' '-DPYTHON_EXECUTABLE=WORKING_DIR
          p/opt/spack/linux-centos7-x86_64/gcc-5.4.0/python-2.7.13-52dgd34kwi53
          hcwhmkggokoplqxzzmfp/bin/python2.7'
  >> 3    CMake Error: The source directory "WORKING_DIR/var/spack/stag
          e/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/nlopt-2.4.2" does not
          appear to contain CMakeLists.txt.
     4    Specify --help for usage, or press the help button on the CMake GUI.

See build log for details:
  WORKING_DIR/var/spack/stage/nlopt-2.4.2-wei4zixpd7mb2b7zn2uniak5svelp6wb/nlopt-2.4.2/spack-build.out

### Information on your system

linux-centos7-x86_64/gcc-5.4.0

So, to be more specific --- I added
```
depends_on('nlopt', type='build')
```
to the r-nloptr package.py and got the following error:
```
==> Installing r-nloptr
==> Using cached archive: WORKING_DIR/var/spack/cache/r-nloptr/r-nloptr-1.0.4.tar.gz
==> Staging archive: WORKING_DIR/var/spack/stage/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/nloptr_1.0.4.tar.gz
==> Created stage in WORKING_DIR/var/spack/stage/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf
==> No patches needed for r-nloptr
==> Building r-nloptr [RPackage]
==> Executing phase: 'install'
==> Error: ProcessError: Command exited with status 1:
    'WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2-qf7qio5ehqhix7yi5epqwovdmoatrg55/bin/R' 'CMD' 'INSTALL' '--library=WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/rlib/R/library' 'WORKING_DIR/var/spack/stage/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/nloptr'

2 errors found in build log:
     61    WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2
           -qf7qio5ehqhix7yi5epqwovdmoatrg55/rlib/R/etc/Makeconf:159: warning:
           ignoring old recipe for target `.f90.o'
     62    WORKING_DIR/lib/spack/env/gcc/g++ -IWORKING_DIR/opt/
           spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2-qf7qio5ehqhix7yi5epqwov
           dmoatrg55/rlib/R/include -DNDEBUG  -I/usr/local/include    -fpic  -g
            -O2  -c dummy.cpp -o dummy.o
     63    WORKING_DIR/lib/spack/env/gcc/gcc -IWORKING_DIR/opt/
           spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2-qf7qio5ehqhix7yi5epqwov
           dmoatrg55/rlib/R/include -DNDEBUG  -I/usr/local/include   -IWORKING_DIR
           s/spack/wip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-4rr
           glbb4xtjqvporzruh36yaxuuhuto2/include   -fpic  -g -O2  -c nloptr.c -
           o nloptr.o
     64    WORKING_DIR/lib/spack/env/gcc/g++ -shared -LWORKING_DIR
           wip/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2-qf7qio5ehqhix7y
           i5epqwovdmoatrg55/rlib/R/lib -L/usr/local/lib64 -o nloptr.so dummy.o
            nloptr.o -LWORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5
           .4.0/nlopt-2.4.2-4rrglbb4xtjqvporzruh36yaxuuhuto2/lib -lnlopt -lm -L
           WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-3.3.2
           -qf7qio5ehqhix7yi5epqwovdmoatrg55/rlib/R/lib -lR
     65    /usr/bin/ld: WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-
           5.4.0/nlopt-2.4.2-4rrglbb4xtjqvporzruh36yaxuuhuto2/lib/libnlopt.a(ge
           neral.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be
            used when making a shared object; recompile with -fPIC
     66    WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2
           .4.2-4rrglbb4xtjqvporzruh36yaxuuhuto2/lib/libnlopt.a: could not read
            symbols: Bad value
  >> 67    collect2: error: ld returned 1 exit status
  >> 68    make: *** [nloptr.so] Error 1
     69    ERROR: compilation failed for package ‘nloptr’
     70    * removing ‘WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc
           -5.4.0/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/rlib/R/librar
           y/nloptr’

See build log for details:
  WORKING_DIR/var/spack/stage/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/nloptr/spack-build.out
```
The detail log ends with:
```
/usr/bin/ld: WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-4rrglbb4xtjqvporzruh36yaxuuhuto2/lib/libnlopt.a(general.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/nlopt-2.4.2-4rrglbb4xtjqvporzruh36yaxuuhuto2/lib/libnlopt.a: could not read symbols: Bad value
collect2: error: ld returned 1 exit status
make: *** [nloptr.so] Error 1
ERROR: compilation failed for package ‘nloptr’
* removing ‘WORKING_DIR/opt/spack/linux-centos7-x86_64/gcc-5.4.0/r-nloptr-1.0.4-l3qyfzbiju6qosl5kqnzdjojgurhbdjf/rlib/R/library/nloptr’
```

--- a/var/spack/repos/builtin/packages/nlopt/package.py
+++ b/var/spack/repos/builtin/packages/nlopt/package.py
@@ -32,11 +32,11 @@ class Nlopt(CMakePackage):
     other algorithms."""
 
     homepage = "https://nlopt.readthedocs.io"
-    url      = "https://github.com/stevengj/nlopt/releases/download/nlopt-2.4.2/nlopt-2.4.2.tar.gz"
+    url      = "https://github.com/stevengj/nlopt/archive/v2.5.0.tar.gz"
     git      = "https://github.com/stevengj/nlopt.git"
 
     version('develop', branch='master')
-    version('2.4.2', 'd0b8f139a4acf29b76dbae69ade8ac54')
+    version('2.5.0', 'ada08c648bf9b52faf8729412ff6dd6d')
 
     variant('shared', default=True, description='Enables the build of shared libraries')
     variant('python', default=True, description='Build python wrappers')

